functions/Restart-DbaService.ps1

```diff
@@ -115,17 +115,27 @@ function Restart-DbaService {
     end {
         $processArray = [array]($processArray | Where-Object { (!$InstanceName -or $_.InstanceName -in $InstanceName) -and (!$Type -or $_.ServiceType -in $Type) })
         foreach ($service in $processArray) {
-            if ($Force -and $service.ServiceType -eq 'Engine' -and !($processArray | Where-Object { $_.ServiceType -eq 'Agent' -and $_.InstanceName -eq $service.InstanceName -and $_.ComputerName -eq $service.ComputerName })) {
-                Write-Message -Level Verbose -Message "Adding Agent service to the list for service $($service.ServiceName) on $($service.ComputerName), since -Force has been specified"
-                #Construct parameters to call Get-DbaService
-                $serviceParams = @{
-                    ComputerName = $service.ComputerName
-                    InstanceName = $service.InstanceName
-                    Type         = 'Agent'
+            if ($Force -and $service.ServiceType -eq 'Engine') {
+                $dependentServices = @()
+                foreach ($dependentService in @("Agent", "PolyBase", "Launchpad")) {
+                    if (!($processArray | Where-Object { $_.ServiceType -eq $dependentService -and $_.InstanceName -eq $service.InstanceName -and $_.ComputerName -eq $service.ComputerName })) {
+                        Write-Message -Level Verbose -Message "Adding $dependentService service to the list for service $($service.ServiceName) on $($service.ComputerName), since -Force has been specified"
+                        $dependentServices += $dependentService
+                    }
+                }
+                if ($dependentServices.Count -gt 0) {
+                    #Construct parameters to call Get-DbaService
+                    $serviceParams = @{
+                        ComputerName  = $service.ComputerName
+                        InstanceName  = $service.InstanceName
+                        Type          = $dependentServices
+                        WarningAction = 'SilentlyContinue'
+
+                    }
+                    if ($Credential) { $serviceParams.Credential = $Credential }
+                    if ($EnableException) { $serviceParams.EnableException = $EnableException }
+                    $processArray += @(Get-DbaService @serviceParams)
```

Comment:
Considering how this command can be used I would consider doing this in a different manner for performance. Using `+=` can build on memory usage for a session in PowerShell.

I'm aware this pattern is used throughout our module, and appears especially in this command; but arrays in PowerShell are immutable so this process basically duplicates that array each time making a larger array object to hold the additional elements. Not something we have to deal with in this PR right now, but something to consider going forward that we need to stop using `+=` in our code for this module.

Reply:
Reference docs that speak on this as well: https://docs.microsoft.com/en-us/windows-server/administration/performance-tuning/powershell/script-authoring-considerations#array-addition

Reply:
I always wondered how is it worth a trouble when it comes to using arrays with less than 10 members. I would understand the necessity if it was an array with 100+ items.
I agree that we should conform to best practices more though. And just like you said, our code is plagued with this particular syntax.

Reply:
Well it is worth the trouble in that to build the objects for those 10 members only takes 1ms compared to 8ms because it was duplicating that array multiple times. Then you have that command run against 100 servers...it can make huge differences; particularly being that this command is likely used in build/deployment cycles frequently.

Reply:
Thanks for the info about array addition. I can definitely see the point, specially when things are at a large scale. I'll try not to use any additional `+=` when submitting future PRs!